### ISSUE TITLE
bodyEl.scrollTop is not a function error

### ISSUE THREAD

**Issue by manar-mk** [harm-less/angular-sticky#18]
Hi there. I am using webpack and angular 1.5
sometimes your plugin throws the 
`bodyEl.scrollTop is not a function error` error. How to fix that?

**Comment by harm-less** [comment]
Interesting, I see [this](http://stackoverflow.com/questions/21556752/jquery-scrolltop-not-working-on-body-element-in-firefox) topic about this on Stack. Perhaps that's the problem. Which browser are you using?

**Comment by AesSedai** [comment]
@harm-less I'm getting the same error in Chrome, Chrome Canary, Firefox, and Safari. Using angular 1.5.8 and angular-sticky v0.3.0. Error is as follows:
```
angular.self-7f8df3e….js?body=1:13921 TypeError: bodyEl.scrollTop is not a function
    at sticksAtPositionTop (angular-sticky.self-fba75ca….js?body=1:228)
    at sticksAtPosition (angular-sticky.self-fba75ca….js?body=1:215)
    at render (angular-sticky.self-fba75ca….js?body=1:242)
    at Object.$api.draw (angular-sticky.self-fba75ca….js?body=1:427)
    at angular-sticky.self-fba75ca….js?body=1:581
    at Object.forEach (angular.self-7f8df3e….js?body=1:322)
    at Object.$sticky.draw (angular-sticky.self-fba75ca….js?body=1:580)
    at angular-sticky.self-fba75ca….js?body=1:512
    at Object.forEach (angular.self-7f8df3e….js?body=1:336)
    at draw (angular-sticky.self-fba75ca….js?body=1:511)
```

**Comment by hkhandare** [comment]
+1

**Comment by eavary** [comment]
Perhaps I'm missing something, but it looks like this issue is resolved by changing:
`bodyEl.scrollTop()`
to:
`bodyEl[0].scrollTop`

**Comment by KaleRoberts** [comment]
Yes, you can resolve this issue by changing bodyEl.scrollTop() to bodyEl[0].scrollTop
However, make sure you're also including jQuery in your project because this is not a pure angular plug-in unfortunately.

**Comment by eavary** [comment]
Thanks for the input... I just went ahead and forked the project and removed jQuery dependency (much like the pull requests on this project do).  It has been working perfectly well without jQuery.

**Comment by henriquecholo** [comment]
Do you guys have another version of this in bower? I can not change the vendor in my project.

**Comment by codeuniquely** [comment]
Still an issue -

webpack:///./~/angular/angular.js?:14362 TypeError: bodyEl.scrollTop is not a function
    at sticksAtPositionTop (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:227)
    at sticksAtPosition (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:214)
    at render (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:241)
    at Object.$api.draw (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:426)
    at eval (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:580)
    at Object.forEach (webpack:///./~/angular/angular.js?:357)
    at Object.$sticky.draw (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:579)
    at eval (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:511)
    at Object.forEach (webpack:///./~/angular/angular.js?:371)
    at draw (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:510)
    at resize (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:507)
    at later (webpack:///./~/angular-sticky-plugin/dist/angular-sticky.js?:681)
    at eval (webpack:///./~/angular/angular.js?:19964)
    at completeOutstandingRequest (webpack:///./~/angular/angular.js?:6131)
    at eval (webpack:///./~/angular/angular.js?:6411)

Is a new version going to appear, which makes a call with the [0] and/or checks for existence of scollTop before calling it ?

**Comment by mazavr** [comment]
I'm using **jqLite** in my angular project. So `bodyEl[0].scrollTop` is not available for that case.

**Comment by aplocher** [comment]
I think the requirements on the [npm page](https://www.npmjs.com/package/angular-sticky-plugin) should be updated to reflect the dependency on JQuery.

**Comment by aplocher** [comment]
If you're using WebPack and you have JQuery, but it's not loading (or loading too late), this is the solution I needed to fix the problem:

https://github.com/l-lin/angular-datatables/issues/993

```
new webpack.ProvidePlugin({
  $: "jquery",
  jQuery: "jquery",
  "window.jQuery": "jquery"
})
```

**Comment by codeuniquely** [comment]
Why not just update teh code so that when you want this value you always just call function call ...

````
scrollTop(elem) {
  if (typeof elem.scrollTop === 'function') {
    return elem.scrolltop();
  } else if (elem.constructor === Array && typeof elem.scrolltop !== 'undefined') {
    return elem[0].scrolltop;
  }
  return (window.pageYOffset !== undefined) ? window.pageYOffset : (document.documentElement || document.body.parentNode || document.body).scrollTop;
}
````
Then you can call it like this when you need to and it'll work with JQuery being present and also when its not there ....

```
var offsetY = scrollTop(bodyEl);
```

**Comment by harm-less** [comment]
Hello everyone,
It seems like I let all of you down by claiming it was a native Angular plugin. I'm sorry for the inconvenience I've caused.

However, I think I just fixed it as the [codepen](http://codepen.io/anon/pen/KgExAR) from @AesSedai works now (it was using the .js from master).
In release [0.4.0](https://github.com/harm-less/angular-sticky/releases/tag/v0.4.0) you'll find the code you're looking for.

Let me know if particular problems still remain. I'm closing this issue for now.
